### PR TITLE
Bug 1139811 - [Cost Control] The device can't pop up the warn box when t...

### PR DESCRIPTION
**Pull request by gitmai**
...he data usage exceeds limit value.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e8126e084ce0b381b6c86c89f396c47658c98c92

File: apps/costcontrol/js/settings/limitdialog.js [hunk 1]
```diff
@@ -33,8 +34,10 @@ function dataLimitConfigurer(guiWidget, settings, viewManager, widgetRoot) {
       var dataLimit = Common.getDataLimit({'dataLimitValue': value,
                                            'dataLimitUnit': currentUnit});
       SimManager.requestDataSimIcc(function(dataSim) {
-        addNetworkUsageAlarm(Common.getDataSIMInterface(dataSim.iccId),
-                             dataLimit);
+        var simInterface = Common.getDataSIMInterface(dataSim.iccId);
+        var self = this;
+        addNetworkUsageAlarm(simInterface, dataLimit, null,
+                             checkDataLimit.call(self, dataLimit));
```
Comment (lodr) [inline apps/costcontrol/js/settings/limitdialog.js]: This is not a function but a call. if you would want to provide a bound-to-self version your should use `.bind(self, dataLimit)`. Notice `this` is undefined in the context the function is passed. What you need is to bind the function in `requestDataSimIcc()` call to bind, then simply use bind with `this` or an arrow function.

File: apps/costcontrol/js/message_handler.js [hunk 2]
```diff
@@ -133,8 +133,8 @@
   }
   window.getTopUpTimeout = getTopUpTimeout;
 
-  function addNetworkUsageAlarm(dataInterface, dataLimit, callback) {
-    NetworkUsageAlarm.updateAlarm(dataInterface, dataLimit, callback);
+  function addNetworkUsageAlarm(dataInterface, dataLimit, onsuccess, onerror) {
+    NetworkUsageAlarm.updateAlarm(dataInterface, dataLimit, onsuccess, onerror);
```
Comment (lodr) [inline apps/costcontrol/js/message_handler.js]: I'm not sure if passing `null` as callback may cause problems. Double check it and if it does, pass an empty function instead. After using promises I started to hate callbacks, ;)

File: apps/costcontrol/js/settings/limitdialog.js [hunk 3]
```diff
@@ -56,6 +59,21 @@ function dataLimitConfigurer(guiWidget, settings, viewManager, widgetRoot) {
...
 
+  function checkDataLimit(dataLimit) {
+    CostControl.getInstance(function _onCostControl(costcontrol) {
+      var requestObj = {
+        type: 'datausage'
+      };
+      costcontrol.request(requestObj, function _onDataStats(result) {
+        var stats = result.data.mobile.total;
+        if (stats > dataLimit) {
+          var message = _('data-limit-notification-text2');
```
Comment (lodr) [inline apps/costcontrol/js/settings/limitdialog.js]: Notice Juwei said [you should show the same message as in the notification](https://bugzilla.mozilla.org/show_bug.cgi?id=1139811#c33) which includes the part of `XX of data used`. You're missing this part right now.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ab1cc19b3065ecde5d534af107087bf400a01560

File: apps/costcontrol/js/settings/limitdialog.js [hunk 4]
```diff
@@ -33,8 +34,10 @@ function dataLimitConfigurer(guiWidget, settings, viewManager, widgetRoot) {
       var dataLimit = Common.getDataLimit({'dataLimitValue': value,
                                            'dataLimitUnit': currentUnit});
       SimManager.requestDataSimIcc(function(dataSim) {
-        addNetworkUsageAlarm(Common.getDataSIMInterface(dataSim.iccId),
-                             dataLimit);
+        var simInterface = Common.getDataSIMInterface(dataSim.iccId);
+        var self = this;
```
Comment (lodr) [inline apps/costcontrol/js/settings/limitdialog.js]: nit: remove `self` as it's not used.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=95d500dc39974b5fd2dc3c17f90a25d86c37ce92

**Pull request merged**